src/conn.c

```diff
@@ -2773,58 +2735,33 @@ natsConn_processMsg(natsConnection *nc, char *buf, int bufLen)
 
     if (!ctrlMsg)
     {
-        sub->msgList.msgs++;
-        sub->msgList.bytes += bufLen;
-
-        if (((sub->msgsLimit > 0) && (sub->msgList.msgs > sub->msgsLimit))
-            || ((sub->bytesLimit > 0) && (sub->msgList.bytes > sub->bytesLimit)))
-        {
-            natsMsg_Destroy(msg);
+        // Do this before we attempt to enqueue the message, even if it were to fail.
+        msg->sub = sub;
+        if ((jsi != NULL) && jsi->ackNone)
+            natsMsg_setAcked(msg);
 
-            sub->dropped++;
-
-            sc = !sub->slowConsumer;
-            sub->slowConsumer = true;
-
-            // Undo stats from above.
-            sub->msgList.msgs--;
-            sub->msgList.bytes -= bufLen;
-        }
-        else
+        if (s == NATS_OK)
+            s = natsSub_enqueueMsg(sub, msg);
```

Comment:
`natsSub_enqueueMsg` - main function for posting messages to dispatch queue, see dispatch.c